charms/kfp-schedwf/src/charm.py

```diff
@@ -76,7 +50,7 @@ def __init__(self, *args):
                 timezone=self.model.config["timezone"],
                 log_level=self.model.config["log-level"],
             ),
-            depends_on=[self.kubernetes_resources],
+            depends_on=[],
```

Comment:
issue: this should depend on leadership component. Before , it was implicitly depending on it by depending on k8s component and k8s component depending on leadership.